### PR TITLE
chore: customer expansion to connections

### PR DESCRIPTION
**Pull request by tomkit**

### CODE REVIEW

File: packages/core/mappers/customer.ts [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
+  const expandConnections = expandedAssociations.includes('connections');
   return {
     id,
     applicationId,
     createdAt,
     updatedAt,
-  } as Customer;
+    connections:
+      expandConnections && connections ? connections.map((connection) => fromConnectionModel(connection)) : undefined,
+  } as CustomerExpanded;
```
Comment (albertyfwu) [inline packages/core/mappers/customer.ts:17]: Can we avoid type assertions?

Comment (albertyfwu) [reply packages/core/mappers/customer.ts:17]: Also, if we return `CustomerExpanded` but the return type of this function is `Customer`, then we don't get any more type-checking.

File: packages/core/mappers/customer.ts [hunk 2]
```diff
@@ -1,10 +1,18 @@
-import type { Customer, Customer as CustomerModel } from '@supaglue/db';
+import type { Customer } from '@supaglue/db';
+import { CustomerExpanded } from '../types/customer';
+import { fromConnectionModel } from './connection';
 
-export const fromCustomerModel = ({ id, applicationId, createdAt, updatedAt }: CustomerModel): Customer => {
+export const fromCustomerModel = (
+  { id, applicationId, createdAt, updatedAt, connections }: CustomerExpanded,
```
Comment (asdfryan) [inline packages/core/mappers/customer.ts:6]: This method signature doesn't make sense to me. Based on the name it looks like you are converting from prisma model to the entity model, but `CustomerExpanded` is not the prisma model.

File: packages/core/mappers/customer.ts [hunk 3]
```diff
@@ -1,10 +1,18 @@
...
+  expandedAssociations: string[] = []
```
[the remark this answers is not in the document]
Comment (tomkit) [reply packages/core/mappers/customer.ts:7]: The same thing in our application, in Prisma, expanded relations

File: packages/core/mappers/customer.ts (hunk 2 above)
Comment (tomkit) [reply packages/core/mappers/customer.ts:6]: +1, imported the wrong Customer

File: packages/core/mappers/customer.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (asdfryan) [reply packages/core/mappers/customer.ts:7]: What I mean is associations is a product terminology -- I think this method should just pass a boolean instead for whether or not to expand connections.

File: packages/core/mappers/customer.ts (hunk 2 above)
Comment (tomkit) [reply packages/core/mappers/customer.ts]: Good catch

File: packages/core/mappers/customer.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (tomkit) [reply packages/core/mappers/customer.ts]: We'll assume we only need 1-hop joins now and have all relations joined if `includeRelations` is set to true

**Review by asdfryan — APPROVED** [review]

**Pull request merged**